src/constants.ts

```diff
@@ -51,3 +51,6 @@ export const AMM_TYPES = {
   "5ocnV1qiCgaQR8Jb8xWnVbApfaygJ8tNoZfgPwsgx9kx": "Sanctum S",
   Gswppe6ERWKpUTXvRPfXdzHhiCyJvLadVvXGfdpBqcE1: "GuacSwap",
 };
+
+export const SWAP_INSTRUCTION_STACK_HEIGHT = 2;
```

Comment:
any reason to check on the stack height? i don't think it should be the case.

Reply:
In some cases if there other AMM instructions (other than swap, may be self call or event) in the inner instructions, then they are also being considered as swap instructions because of this condition
` instruction.programId.toBase58() in AMM_TYPES`. So to avoid this I have used the stackHeight

Reply:
i don't think `stack height = 2` covers all the cases tho.